### ISSUE TITLE
false positive in `prefer_asserts_in_initializer_lists`

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#1557]
triggering test case:

```dart
class F<T> {
  bool get foo => true;
}

class G<T> extends F<T> {
  G()
    {
      assert(foo); // OK
    }
}
```
